source/NVDAObjects/IAccessible/__init__.py

```diff
@@ -1560,11 +1560,24 @@ def _get_isPresentableFocusAncestor(self):
 class TrayClockWClass(IAccessible):
 	"""
 	Based on NVDAObject but the role is changed to clock.
+	The value contains the time, and the date if available.
 	"""
 
 	def _get_role(self):
 		return controlTypes.ROLE_CLOCK
 
+	def _get_name(self):
+		#The name contains redundant information that is available either in the role or the value.
+		return None
+
+	def _get_value(self):
+		value = super(TrayClockWClass, self).value
+		if value is None:
+			#On XP there is no value, fall back to using the name.
+			return super(TrayClockWClass, self).name
+		#Strip Unicode left-to-right and right-to-left marks.

```

Comment:
I think referencing a ticket might help future devs to find out why the below code is necessary.